src/conn_sock.c

```diff
@@ -151,6 +152,75 @@ static char *setup_socket(int *fd, const char *path)
 
 	return csname;
 }
+#endif
+#ifdef __FreeBSD__
+static char *setup_socket(int *fd, const char *path)
+{
+	struct sockaddr_un addr = {0};
+	char *csname = NULL;
+	_cleanup_close_ int sfd = -1;
+
+	if (path != NULL) {
+		_cleanup_free_ char *dname_buf = NULL;
+		_cleanup_free_ char *bname_buf = NULL;
+		char *dname = NULL, *bname = NULL;
+
+		csname = strdup(path);
+		dname_buf = strdup(path);
+		bname_buf = strdup(path);
+		if (csname == NULL || dname_buf == NULL || bname_buf == NULL) {
+			pexit("Failed to allocate memory");
+			return NULL;
+		}
+		dname = dirname(dname_buf);
+		if (dname == NULL)
+			pexitf("Cannot get dirname for %s", csname);
+
+		sfd = open(dname, O_CREAT, 0600);
+		if (sfd < 0)
+			pexit("Failed to create file for console-socket");
+
+		bname = basename(bname_buf);
+		if (bname == NULL)
+			pexitf("Cannot get basename for %s", csname);
+
+		strncpy(addr.sun_path, bname, sizeof(addr.sun_path) - 1);
+	} else {
+		_cleanup_free_ const char *tmpdir = g_get_tmp_dir();
+
+		csname = g_build_filename(tmpdir, "conmon-term.XXXXXX", NULL);
+		/*
+		 * Generate a temporary name. Is this unsafe? Probably, but we can
+		 * replace it with a rename(2) setup if necessary.
+		 */
+		int unusedfd = g_mkstemp(csname);
+		if (unusedfd < 0)
+			pexit("Failed to generate random path for console-socket");
+		close(unusedfd);
+		/* XXX: This should be handled with a rename(2). */
+		if (unlink(csname) < 0)
+			pexit("Failed to unlink temporary random path");
+
+		strncpy(addr.sun_path, csname, sizeof(addr.sun_path) - 1);
+	}
+
+	addr.sun_family = AF_UNIX;
+	ndebugf("addr{sun_family=AF_UNIX, sun_path=%s}", addr.sun_path);
+
+	/* Bind to the console socket path. */
+	*fd = socket(AF_UNIX, SOCK_STREAM | SOCK_CLOEXEC, 0);
+	if (*fd < 0)
+		pexit("Failed to create socket");
+	if (bindat(sfd == -1 ? AT_FDCWD : sfd, *fd, (struct sockaddr *)&addr, sizeof(addr)) < 0)
+		pexit("Failed to bind to console-socket");
+	if (fchmodat(sfd, addr.sun_path, 0700, AT_SYMLINK_NOFOLLOW))
+		pexit("Failed to change console-socket permissions HERE!!");
```

Comment:
hm is this HERE!! leftover from debugging

Reply:
> hm is this HERE!! leftover from debugging

Oops - I'll get rid of it.